### PR TITLE
controller: Move StorageMaximum from disk.spec to disk.status

### PR DESCRIPTION
**Pull request by JacieChao**
https://github.com/rancher/longhorn/issues/212

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -165,3 +165,8 @@ func GetDiskConditionFromStatus(status DiskStatus, conditionType DiskConditionTy
 	}
 	return condition
 }
+
+type DiskUpdateInfo struct {
+	DiskSpec
+	StorageMaximum int64 `json:"storageMaximum"`
```
Comment (yasker) [inline types/types.go]: I think we should able to set `ReservedSpace` instead of maximum directly for the testing? What do you think?

Comment (JacieChao) [reply types/types.go]: Yeah, that sounds nice and more like user operation. I think it maybe more difficult to test edge situation of scheduler. We need to recalculate reserved space for disks in different cluster.

Comment (yasker) [reply types/types.go]: I see. But it sounds weird to have the option in the status but an API able to update it (and user suppose not to update it by themselves). I think it makes sense to do a bit extra work on the calculation.

Comment (JacieChao) [reply types/types.go]: What about remove update maximum field from API in this release? I will dig more in longhorn-test later.
